### DISCUSSION TITLE
Where do I find functions' source code

### DISCUSSION THREAD

**Discussion by aleemont** [fish-shell/fish-shell#9184] [Q&A] [answered]
I accidentally removed alias function. I want to implement it again, any help?

**Comment by faho** [comment, marked as answer]
If you mean you removed fish's stock `alias` function: https://github.com/fish-shell/fish-shell/blob/master/share/functions/alias.fish, or reinstall your package.

Note that if you fiddled in /usr/share/fish/functions (where the function is stored by default): Don't, that's a terrible idea. If you instead saved a new version of `alias`: Just delete that file, e.g. `~/.config/fish/functions/alias.fish`. Fish looks up functions by going through the directories in $fish_function_path and picks the first matching one. If you saved a new version it'll just shadow the other file.

If you mean you removed a specific function: That's up to you. Fish stores functions when you tell them to, for example by running `funcsave`, which stores them in ~/.config/fish/functions/$functionname.fish. If you didn't store it, or you deleted that file, or removed the line from the config: Sorry, there's nothing we can do. If you did it interactively it's possibly still in history, do `history search`.
